### PR TITLE
Web socket for java

### PR DESCRIPTION
**Pull request by RavivGuy**

### CODE REVIEW

File: quoridor-logic-server/src/main/java/com/rl/mpquoridor/controllers/GameWebSocket.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import java.util.List;
+
+@Controller
+public class GameWebSocket {
+
+    @Autowired
+    private SimpMessagingTemplate messageSender;
+
+
+    @MessageMapping("/movePawn/{gameId}")
```
Comment (Guyeise1) [inline quoridor-logic-server/src/main/java/com/rl/mpquoridor/controllers/GameWebSocket.java:20]: The order should be opposite.
{gameId}/movePawn

or if its not possible then
/game/{gameId}/movePawn

File: quoridor-logic-server/src/main/java/com/rl/mpquoridor/controllers/GameWebSocket.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    public void movePawn(@PathVariable String gameId, TurnAction action) {
```
Comment (Guyeise1) [inline quoridor-logic-server/src/main/java/com/rl/mpquoridor/controllers/GameWebSocket.java:21]: TurnAction -> MovePawn

File: quoridor-logic-server/src/main/java/com/rl/mpquoridor/controllers/GameWebSocket.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        // todo: play the movePawn and return the game status
+        this.messageSender.convertAndSend("/topic/gameStatus/" + gameId, action);
+    }
+
+    @MessageMapping("/putWall/{gameId}")
+    public void putWall(@PathVariable String gameId, TurnAction action) {
```
Comment (Guyeise1) [inline quoridor-logic-server/src/main/java/com/rl/mpquoridor/controllers/GameWebSocket.java:27]: TurnAction -> PlaceWall

**Review by Guyeise1 — CHANGES_REQUESTED** [review]
There are lots of files here that I have different PR with some changes.
Those files should not be a part of this PR.

File: quoridor-logic-server/src/main/java/com/rl/mpquoridor/models/GameResult.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import com.rl.mpquoridor.models.players.Player;
+import com.sun.istack.NotNull;
+import org.springframework.data.util.Pair;
+
+import java.util.Collections;
+import java.util.List;
+
+public class GameResult {
+    private final Player winner;
+    private final List<Pair<Player, TurnAction>> history;
```
Comment (shvshv44) [inline quoridor-logic-server/src/main/java/com/rl/mpquoridor/models/GameResult.java:13]: I think its more clean to make 'Pair' field into a pojo contains two fields.
history could be of type PlayerTurnData, or HistoryMove or something with the same context

File: quoridor-logic-server/src/main/java/com/rl/mpquoridor/models/GameResult.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+    public GameResult(@NotNull Player winner, List<Pair<Player, TurnAction>> history) {
+        this.winner = winner;
+        this.history = history;
+    }
+
+    public Player getWinner() {
```
Comment (shvshv44) [inline quoridor-logic-server/src/main/java/com/rl/mpquoridor/models/GameResult.java:20]: You could use Lombok like Sharon did.
Its working fine with pojos

File: quoridor-logic-server/src/main/java/com/rl/mpquoridor/controllers/GameWebSocket.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+
+    public void endTurn(String gameId, TurnAction action) {
+        this.messageSender.convertAndSend("/topic/gameStatus/" + gameId, action);
+    }
+
+
+    public void avaliableMoves(String gameId, List<Position> positions) {
+        // send the available moves of the next turn
```
Comment (SharonEizner) [inline quoridor-logic-server/src/main/java/com/rl/mpquoridor/controllers/GameWebSocket.java:39]: i dont think that u need to send this. 
in the client u have basic validation, and in the server u will get error message if u cant put the wall

**Review by shvshv44 — APPROVED** [review]

**Review by shvshv44 — APPROVED** [review]

**Review by SharonEizner — APPROVED** [review]

**Review by shvshv44 — APPROVED** [review]

**Review by shvshv44 — APPROVED** [review]

**Review by shvshv44 — APPROVED** [review]

**Pull request merged**